src/app/features/charts/components/median-barchart/median-barchart.component.spec.ts

```diff
@@ -0,0 +1,222 @@
+    const { chart } = setUp([]);
+
+    expect(component.data?.length).toEqual(0);
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should not render the chart if all values are negative', () => {
+    const { chart } = setUp(
+      SMALL_MOCK_DATA.map((obj) => {
+        return { ...obj, medianlogcpm: -1 * obj.medianlogcpm };
+      })
+    );
+
+    expect(element.querySelector('.chart-no-data')).toBeTruthy();
+
+    expect(chart).toBeFalsy();
+  });
+
+  it('should render the chart if there is positive data', () => {
+    const createChartSpy = spyOn(component, 'createChart').and.callThrough();
+    const { chart } = setUp();
+
+    expect(component.data?.length).toEqual(FULL_MOCK_DATA.length);
+    expect(createChartSpy).toHaveBeenCalled();
+    expect(chart).toBeTruthy();
+
+    expect(element.querySelector('.chart-no-data')).toBeFalsy();
+  });
+
+  it('should render the chart axes', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelector('.x-axis')).toBeTruthy();
+    expect(chart?.querySelector('.y-axis')).toBeTruthy();
+  });
+
+  it('should render the correct number of bars', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      FULL_MOCK_DATA.length
+    );
+  });
+
+  it('should not render bars for negative values', () => {
+    const { chart } = setUp([
+      { ...SMALL_MOCK_DATA[0], medianlogcpm: -0.01 },
+      ...SMALL_MOCK_DATA.slice(1),
+    ]);
+
+    expect(chart?.querySelectorAll('rect').length).toEqual(
+      SMALL_MOCK_DATA.length - 1
+    );
+  });
+
+  it('should render the bar labels', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelectorAll('text.bar-labels').length).toEqual(
+      FULL_MOCK_DATA.length
+    );
+  });
+
+  it('should render the labels for both axes', () => {
+    const { chart } = setUp();
+
+    expect(chart?.querySelector('.x-axis-label')?.textContent).toEqual(
+      XAXIS_LABEL
+    );
+    expect(chart?.querySelector('.y-axis-label')?.textContent).toEqual(
+      YAXIS_LABEL
+    );
+  });
+
+  it('should render the meaningful expression threshold', () => {
+    const { chart } = setUp();
+
+    expect(
+      chart?.querySelector('.meaningful-expression-threshold-line')
+    ).toBeTruthy();
+  });
+
+  it('should render the meaningful expression threshold even when all values are small', () => {
+    const { chart } = setUp(SMALL_MOCK_DATA);
+
+    expect(
+      chart?.querySelector('.meaningful-expression-threshold-line')
+    ).toBeTruthy();
+  });
+
+  it('should alphabetize the x-axis values', () => {
+    setUp(SMALL_MOCK_DATA);
+
+    const sortedTissues = TISSUES.sort();
+    const xAxisTicks = element
+      .querySelector('svg g .x-axis')
+      ?.querySelectorAll('.tick');
+    xAxisTicks?.forEach((val, index) => {
+      expect(val.textContent).toEqual(sortedTissues[index]);
+    });
+  });
```

Comment:
Ensure new implementation matches current behavior, which alphabetizes the x-axis values